### PR TITLE
hotfix/JM-7862 Juror deferral issue

### PR DESCRIPTION
**Pull request by akikrahman1**
### JIRA link (if applicable) ###

https://centralgovernmentcgi.atlassian.net/browse/JM-7862

### Change description ###

Updated code to remove any queued deferral/excusal letters when marking as responded

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/juror/api/moj/service/PrintDataServiceImpl.java [hunk 1]
```diff
@@ -229,6 +232,21 @@ public void checkLetterInBulkPrint(String jurorNumber, String formType, LocalDat
...
 
+    @Override
+    public void removeQueuedLetterForJuror(JurorPool jurorPool) {
+
+        // determine form codes for juror status
+        List<FormCode> formCodes = FormCode.getFormCodesForJurorStatus(jurorPool.getStatus().getStatus());
+
+        List<BulkPrintData> bulkPrintDataList = bulkPrintDataRepository
+            .findByJurorNoAndCreationDateAndFormAttributeFormTypeIn(
+            jurorPool.getJuror().getJurorNumber(), LocalDate.now(),
```
Comment (Ben-Edwards-cgi) [inline src/main/java/uk/gov/hmcts/juror/api/moj/service/PrintDataServiceImpl.java:243]: Do we need to remove the creation date filter?
And get any letters that have not been printed as letters can take between 2 to 4 days to be sent?

File: src/main/java/uk/gov/hmcts/juror/api/moj/service/PrintDataServiceImpl.java [hunk 2]
```diff
@@ -229,6 +232,21 @@ public void checkLetterInBulkPrint(String jurorNumber, String formType, LocalDat
...
+            formCodes.stream().map(FormCode::getCode).toList()
+        );
+
+        bulkPrintDataRepository.deleteAll(bulkPrintDataList);
```
Comment (Ben-Edwards-cgi) [inline src/main/java/uk/gov/hmcts/juror/api/moj/service/PrintDataServiceImpl.java:247]: We should already strip out letters that do not need to be printed apart of the letter Job.

I.e we check juror status and if they have changed we skip that row for printing

File: src/main/java/uk/gov/hmcts/juror/api/moj/service/PrintDataServiceImpl.java (hunk 1 above)
Comment (akikrahman1) [reply src/main/java/uk/gov/hmcts/juror/api/moj/service/PrintDataServiceImpl.java:243]: The specific use case here is the user has mistakenly deferred/excused someone, they now need to mark as responded and dont want any letters queued today, if they miss this chance, then they are aware the letter will go out and nothing they can do about it

File: src/main/java/uk/gov/hmcts/juror/api/moj/service/PrintDataServiceImpl.java (hunk 2 above)
Comment (akikrahman1) [reply src/main/java/uk/gov/hmcts/juror/api/moj/service/PrintDataServiceImpl.java:247]: understood, I think its cleaner to remove the queued letter as its not required at all

**Comment by akikrahman1** [comment]
parking this for now as Liam confirmed client does not need this functionality atm

**Pull request closed without merging**